sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/PowershellManager.java

```diff
@@ -6,173 +6,56 @@
-                    new OutputStreamWriter(new BufferedOutputStream(process.getOutputStream()), StandardCharsets.UTF_8),
-                    true);
-                if (this.process.waitFor(4L, TimeUnit.SECONDS) && !this.process.isAlive()) {
-                    throw new CredentialUnavailableException("Unable to execute PowerShell."
-                        + " Please make sure that it is installed in your system.");
-                }
-                this.closed = false;
-            } catch (InterruptedException | IOException e) {
-                throw new CredentialUnavailableException("Unable to execute PowerShell. "
-                    + "Please make sure that it is installed in your system", e);
-            }
-            return this;
-        };
-        return executorService != null ? Mono.fromFuture(CompletableFuture.supplyAsync(supplier, executorService))
-            : Mono.fromFuture(CompletableFuture.supplyAsync(supplier));
     }
 
-
-    public Mono<String> runCommand(String command) {
-        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(),
-            StandardCharsets.UTF_8));
-        StringBuilder powerShellOutput = new StringBuilder();
-        commandWriter.println(command);
-        return canRead(reader)
-                .flatMap(b -> {
-                    if (b) {
-                        return readData(reader, powerShellOutput)
-                                .flatMap(ignored -> Mono.just(PS_RESPONSE_PATTERN.matcher(powerShellOutput.toString())
-                                        .replaceAll("")));
-                    } else {
-                        return Mono.error(new CredentialUnavailableException("Error reading data from reader"));
-                    }
-                });
-    }
-
-    private Mono<Boolean> readData(BufferedReader reader, StringBuilder powerShellOutput) {
-        return Mono.defer(() -> {
-            String line;
+    public Mono<String> runCommand(String input) {
+        return Mono.fromCallable(() -> {
             try {
-                line = reader.readLine();
-                if (line != null) {
-                    powerShellOutput.append(line).append("\r\n");
-                    return canRead(reader).flatMap(b -> {
-                        if (!this.closed && b) {
-                            return Mono.empty();
-                        }
-                        return Mono.just(true);
-                    });
+                String[] command;
+                if (Platform.isWindows()) {
+                    // Windows PowerShell command
+                    command = new String[]{powershellPath, "-Command", input};
                 } else {
-                    return Mono.just(true);
+                    // Unix-based (macOS/Linux) shell command
+                    command = new String[]{"/bin/bash", "-c", String.format("%s -Command '%s'", powershellPath, input)};
```

Comment:
Does putting it all in a single argument with the format wrap it in quotes or something? Why can't it be the same as above?

Reply:
Refactored to be in a single statement. 
But since we're building a string array, the arguments are different.